src/lib/_internal_sense_hat.js

```diff
@@ -264,16 +264,34 @@ var $builtinmodule = function (name) {
         return compass;
     });
 
-    // Magnetometer tilt compensation
-    // From here http://diydrones.com/forum/topics/magnetometer-tilt-compensation-tutorial-sensor-analysis?commentId=705844%3AComment%3A1657670
     mod.headingRead = new Sk.builtin.func(function () {
-        var x, y, z;
-        x = Sk.sense_hat.rtimu.fusionPose[0];
-        y = Sk.sense_hat.rtimu.fusionPose[1];
-        z = Sk.sense_hat.rtimu.fusionPose[2];
-        jsheading = Math.atan2((z * Math.sin(y) - y * Math.cos(y)), x * Math.cos(x) + y * Math.sin(x) * Math.sin(y) + z * Math.sin(x) * Math.cos(y));
-        var heading = Sk.ffi.remapToPy(jsheading);
-
+        /* Returns tilt-compensated magnetometer heading in radians */
+        /* Note: RTIMULib calculates a moving average. This gives an instant reading */
+
+        // Accelerometer's roll and pitch, used for compensation
+        var x, y;
+        x = Sk.sense_hat.rtimu.fusionPose[0]; // roll
+        y = Sk.sense_hat.rtimu.fusionPose[1]; // pitch
+
+        // Compass raw values in microteslas
+        var mx, my, mz;
+        mx = Sk.sense_hat.rtimu.compass[0];
+        my = Sk.sense_hat.rtimu.compass[1];
+        mz = Sk.sense_hat.rtimu.compass[2];
```

Comment:
By reading those forum posts and some other sources and the sensehat source I had the illusion that we should have used the angles produced by reading only from compass but transformed to orientation. Therefore, I though compas orientationis the same as normal orientation.

But now we are using the compass radians /microteslas for those...